product_assortment/models/ir_filters.py

```diff
@@ -83,17 +81,17 @@ def get_partner_domain_fields(self):
         for ir_filter in self.sudo().search([("is_assortment", "=", True)]):
             domain = ir_filter._get_eval_partner_domain()
             for item in domain:
-                if isinstance(item, (list, tuple)) and isinstance(item[0], str):
+                if isinstance(item, list | tuple) and isinstance(item[0], str):
                     field_set.add(item[0].split(".")[0])
         return field_set
 
     @api.depends("partner_ids", "partner_domain")
     def _compute_all_partner_ids(self):
         """Summarize selected partners and partners from partner domain field"""
-        for ir_filter in self:
+        for ir_filter in self.sudo():
             if not ir_filter.is_assortment:
                 ir_filter.all_partner_ids = False
-            elif ir_filter.partner_domain != "[]":
```

Comment:
Likely this is the big differentiator between the PR's.
`elif ir_filter.partner_domain != "[]":`
would never evaluatue to true. Because by default the value assigned is `[]`

Reply:
@PieterPaulussen Can you agree, that the evaluation in this code is wrong?
(Has been there for quite some versions, but undeteced by unit tests / use cases)